### ISSUE TITLE
It's been 144 blocks, but funds are nowhere.  How do I trace which party sent it to where, and who has it?

### ISSUE THREAD

**Issue by Ulmo** [ElementsProject/lightning#768]
I have two nodes.  I was trying to make a donation to an informative website from one of my nodes, when it failed with some error that there was insufficient fee, so I tried upping the fee on my node, which I think caused a bunch of nodes to ban my node.  Anyway, during all that, I also upgraded both my nodes and restarted them.  Now, I have this odd situation where my node banned my other node, and closed the channel, claiming the funding transaction was spent, but I cannot understand where it went.

Here's the log on my node where I accidentally set the fee too high, grepped for my other node; you can see that it starts counting up depth until depth 131, when I update that process (stop and restart), and now I don't see any more information in the log about that transaction any more.

[excerpt2.txt](https://github.com/ElementsProject/lightning/files/1662523/excerpt2.txt)

The log of that transaction 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180 in the other node goes up to depth 100 then stops counting higher.

I do not see the funds show up anywhere.

Why is the depth no longer counted on either node?  Where are the funds?

**Comment by ZmnSCPxj** [comment]
Needs more analysis beyond what I can offer, I am afraid.  As a first glance it seems we lost track of the fact that the channel has to be redeemed.  We entered ONCHAIN_OUR UNILATERAL, then we stopped, and on restarting we forgot that we were in ONCHAIN_OUR_UNILATERAL and should be monitoring the depth of a particular transaction.

https://github.com/ElementsProject/lightning/blob/master/lightningd/lightningd.c#L308 is where we load channels at startup.  I would assume we should do something special if we are in ONCHAIN_OUR_UNILATERAL at this point.  Grepping `ONCHAIN_OUR_UNILATERAL` shows we do not do much of anything about this state: indeed this line https://github.com/ElementsProject/lightning/blob/b91556283b487df318fbff380e1801c0084dc268/lightningd/peer_control.c#L532 suggests we do not try to return monitoring this state; instead we just wait for the other side to try to do something with that channel and then error the channel to them.

@cdecker may be more helpful here.  From my point of view it looks like a legitimate bug "We do not return to monitoring our-side unilateral close if we are stopped and then restarted".

**Comment by Ulmo** [comment]
If one of the states a channel can be in isn't saved or recovered, then I wonder if there are many such states a channel can be in that aren't saved or recovered, meaning a generalized saving and recovery mechanism would be desired.

**Comment by ZmnSCPxj** [comment]
It is saved in the database; the issue I think is that we are not handling it on startup, only on normal operation.  A "general recovery mechanism" would not help much if on startup we ignore the (already loaded) state the channel is in.

During initialization we have an `activate_peers` call that should begin triggering on each channel, and it should put a watch on the funding transaction if it exists.  Possibly the issue is that this trigger only gets called when we transition from unspent->spent (I think; @cdecker should know more).  On restarting, the funding transaction is already in a "spent" state, and it will just get deeper into a "spent" state.  So it is possible that this is the root cause.  Somebody needs to dig into topology subsystem, I have not explored that part yet.  Probably much safer for `watch_utxo` to trigger at `begin_topology` if it finds the UTXO was spent.

@Ulmo, in the logs *after* restart, can you grep `with no funding txid`?  If we can find it in the log after restart, then we lost track of the funding tx.  Otherwise, it is more likely that we do not trigger `watch_utxo` on startup.

**Comment by ZmnSCPxj** [comment]
Looking a little into topology (and keeping in mind that I am ignorant of much of topology) it seems that `begin_topology` does very very little: indeed all it does is tell topology to update itself from the bitcoind.

Thus, it seems very likely that `watch_txo` only triggers on an unspent->spent edge.

So what happened (I think) is this:

1.  We saw the funding tx output move from unspent->spent.  Topology records this fact into its own db.
2.  We check the tx that spent the funding tx output and identified it as a `ONCHAIND_OUR_UNILATERAL` tx.  We start monitoring that tx is confirmed to depth 144.
3.  We are stopped.  This of course ends the monitoring process.
4.  We are restarted.
5.  We install a watch again on the funding tx output in `activate_peers`.
6.  Topology starts up.  Since the funding tx output is already spent, topology *ignores* the watch on the funding tx output (which should get triggered only on a unspent->spent edge).

There are some possible solutions:

1.  On `begin_topology`, for every `watch_txo`, we should trigger every txo that is known to already be spent.  This will prevent this and similar txo-monitoring bugs in the future, but may be surprising/unexpected behavior for that subsystem.
2.  On `activate_peer`, if we are in `FUNDING_SPENT`, `ONCHAIND_MUTUAL`, `ONCHAIND_OUR_UNILATERAL`, or `ONCHAIND_THEIR_UNILATERAL`, we should call directly into `funding_spent` to restart `onchaind` (and force us back into `FUNDING_SPENT`), which should re-analyze what has happened on chain.  However we need to keep track of what tx exactly spent the funding tx output.  Maybe topology has that data?

Deferring to @cdecker, who might have more accurate information and better view of the situation.

**Comment by cdecker** [comment]
It might be related to #709, could you pull and recompile with `master`? That should allow `lightningd` to sync correctly with the blockchain again.

**Comment by Ulmo** [comment]
1.  ZmnSCPxj, I just got home from (unrelated) work, and grep'd log "with no funding txid" in the node that ONCHAIND_OUR_UNILATERAL'd my other node, and everything listed is neither of my nodes:

lightningd(8760): peer 035f1498c929d4cefba4701ae36a554691f526ff60b1766badd5a49b3c8b68e1d8: activate_peer(OPENINGD) with no funding txid?
lightningd(8760): peer 035f1498c929d4cefba4701ae36a554691f526ff60b1766badd5a49b3c8b68e1d8: activate_peer(OPENINGD) with no funding txid?
lightningd(8760): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(8760): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(8760): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(8760): peer 03204b29fe80bbdb895b50f40fb81a78ccc53f2de8eca6bb234facffdbf783ec96: activate_peer(OPENINGD) with no funding txid?

2.  I tried that grep in the log before the restart, for heck of it, and neither of my nodes are listed there either:
[grep2.txt](https://github.com/ElementsProject/lightning/files/1666292/grep2.txt)

3.  I repeated that on my other node and came up with neither of my nodes.

Ok, so in your logic, that means we didn't necessarily lose track of funding tx, and didn't trigger watch_utxo on start.

Your next message about topology:

You got the sequence I experienced, as far as I understand it, correct.

That seems like a lot of great starting points.

On to cdecker:  I'm looking at #709:

I had trouble finding exact duplicate behavior in my nodes to things described (I only spend 15 minutes trying to compare), but I do have 69ab6ac8534705e623eaf9504de0c6467853b36950bee9c9e8d8d6ad18eb5d2c stuck in CHANNELD_AWAITING_LOCKIN in my less problematic node for many days.

So, I'm pulling down a new copy of master, building, installing and restarting:

What am I looking for?  I'll list my funds:  None.  I looked for my other node:  it only has "node_announcement" and "Read decrypt" log entries.  I looked for "with no funding txid", and only found nothing about my node:

lightningd(20003): peer 035f1498c929d4cefba4701ae36a554691f526ff60b1766badd5a49b3c8b68e1d8: activate_peer(OPENINGD) with no funding txid?
lightningd(20003): peer 035f1498c929d4cefba4701ae36a554691f526ff60b1766badd5a49b3c8b68e1d8: activate_peer(OPENINGD) with no funding txid?
lightningd(20003): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(20003): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(20003): peer 029dbd7baa96252b6f000e3522ab1b81884a43c47d517c5effde0160e67c6961c1: activate_peer(OPENINGD) with no funding txid?
lightningd(20003): peer 03204b29fe80bbdb895b50f40fb81a78ccc53f2de8eca6bb234facffdbf783ec96: activate_peer(OPENINGD) with no funding txid?

In sqlite3, I find the peer:

sqlite> select peer_id,funding_tx_outnum, hex(funding_tx_id), state,funder,channel_flags,hex(closing_sig_received),last_was_revoke from channels where peer_id=2;
2|0|60D4F5831B1230A2C5EA0378CE8884232FC6D58CFBC7CEDCCAAB49BF2208C015|10|0|1||0
sqlite> select id,hex(node_id),address from peers where id=2;
2|031C527494F295220D26A1814C18E7DB3F4F03203CE1A53A0B74B52ED570A3972A|24.4.228.213:9735
sqlite> 

I haven't yet restarted that (better behaved) peer.  I wonder if I should.  Ok, I will, with the update.

Is that transaction id something I'm interested in?  Well, I don't know (the below results seem to make me think otherwise), but I looked it up anyway:

In my better behaved node, in my old log, I found that tx id, with many entries like this:

lightningd(16747): Broadcasting tx 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b9700000000001600140335a6b8c2a870b08eb6363128027042b607c22a040047304402207c5392f4f01eabf571886a07983632bdbc846d084ae5bfb93f9cf148f91eb3bd02205cfffb995bcd36ccb56d37d736e5d5f2123ee312136c5e9148e4461072a2c91001483045022100d33a0e02b51ec14d29a53d85b0cdd5fdecd07600b8a6e06e115ef5b876799cb2022070f028ab2cb88c1e0b9aadef4463630313cffc3ce3738bc0a3f30116435a5e4b01475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620: 25 error code: -25

However, the txid doesn't show up in the new restarted log of my better behaved node.

Ok, back to my worse behaved node:  seems more information is forthcoming:

lightningd(20003): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Got UTXO spend for 15c
00822bf49abcadccec7fb8cd5c62f238488ce7803eac5a230121b83f5d460:0: 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979
da515174fbf5180
lightningd(20003): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Peer permanent failure
 in ONCHAIND_OUR_UNILATERAL: Funding transaction spent
lightningd(20003):       (tx 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
lightningd(20003): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c
015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd354104004
73044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2
cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8
fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c42853
8229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620

Have I seen this before?  At first glance, I think so:
[maybe.txt](https://github.com/ElementsProject/lightning/files/1666435/maybe.txt)

In fact, I've seen that exact transaction being sent before:

ulmo@ubcln:~/log$ egrep '0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620' log.? log
log.3:lightningd(23594): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
log.3:lightningd(23594): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
log.3:lightningd(20087): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
log:lightningd(20003): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
ulmo@ubcln:~/log$ 

(log.? are chronologically before log)

In context, for that exact transaction, the first time:

lightning_channeld(23689): TRACE: Can't send commit: waiting for revoke_and_ack
lightning_channeld(23696): TRACE: Can't send commit: waiting for revoke_and_ack
lightning_channeld(27209): TRACE: Can't send commit: waiting for revoke_and_ack
lightning_channeld(23680): TRACE: Can't send commit: waiting for revoke_and_ack
lightning_channeld(27209): TRACE: Read decrypt 001160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c01
5002d7570646174655f6665652031303030303030206f7574736964652072616e67652033313630302d333035303330
lightning_channeld(27209): TRACE: peer_in WIRE_ERROR
lightning_gossipd(23602): TRACE: Forgetting remote peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed5
70a3972a
lightning_channeld(27209): STATUS_FAIL_PEER_BAD: Peer sent error channel 60d4f5831b1230a2c5ea0378ce8884232fc6d58cf
bc7cedccaab49bf2208c015: update_fee 1000000 outside range 31600-305030
lightningd(23594): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Peer permanent failure
 in CHANNELD_NORMAL: lightning_channeld: Peer sent error channel 60d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedcc
aab49bf2208c015: update_fee 1000000 outside range 31600-305030
lightningd(23594):       (tx 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
lightningd(23594): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
lightning_gossipd(23602): TRACE: Forgetting remote peer 03204b29fe80bbdb895b50f40fb81a78ccc53f2de8eca6bb234facffdbf783ec96
lightning_channeld(23615): TRACE: Can't send commit: waiting for revoke_and_ack
lightning_channeld(23613): TRACE: Read decrypt 00116446dd267f840b53eb5b64e367a464601e99d5d8597a280cd2ef51348b609149002d7570646174655f6665652031303030303030206f7574736964652072616e67652034323234382d323930323135


The second time:

lightningd(23594): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Got UTXO spend for 15c
00822bf49abcadccec7fb8cd5c62f238488ce7803eac5a230121b83f5d460:0: 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979
da515174fbf5180
lightningd(23594): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Peer permanent failure
 in CHANNELD_NORMAL: Funding transaction spent
lightningd(23594):       (tx 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
lightningd(23594): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
lightningd(23594): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: state: CHANNELD_NORMAL -> FUNDING_SPEND_SEEN
lightning_onchaind(3945): pid 3945, msgfd 26
lightningd(23594): Forcing fee rate, ignoring estimate


The third time:

lightningd(20087): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Got UTXO spend for 15c
00822bf49abcadccec7fb8cd5c62f238488ce7803eac5a230121b83f5d460:0: 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979
da515174fbf5180
lightningd(20087): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Peer permanent failure
 in ONCHAIND_OUR_UNILATERAL: Funding transaction spent
lightningd(20087):       (tx 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
lightningd(20087): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
lightningd(20087): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: state: ONCHAIND_OUR_UNILATERAL -> FUNDING_SPEND_SEEN
lightning_onchaind(19639): pid 19639, msgfd 26


And the fourth and last time after this restart:

lightningd(20003): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Got UTXO spend for 15c
00822bf49abcadccec7fb8cd5c62f238488ce7803eac5a230121b83f5d460:0: 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979
da515174fbf5180
lightningd(20003): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: Peer permanent failure
 in ONCHAIND_OUR_UNILATERAL: Funding transaction spent
lightningd(20003):       (tx 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
lightningd(20003): sendrawtransaction: 0200000000010160d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015000000000096eec38001b07b970000000000220020c16c89e21ce34a861142a8e80f203dfc8fabf97fb4017344a8e354866cdd35410400473044022060b2bf77abff87424a82f7d8314878b03024ca9ccda4f516a08bb364b128a1ea0220023b8ffff86264c16f1b055b95b93c00130b2cf42200e073e7b859bdd36b068301483045022100d46a5c7eb1a500125369885451a37600a8f7f04a0059e46b21aa0d7970f3a8b10220274a8fd4b2e99147039d34b0296165eb18e07f2c66daea14877fd624cae19c1001475221028336767380e4c0d9d1e93656432712aee1eaff1c428538229d8412f27d2276932103f202c24f865fc73afd3b55b0e851233c06f3cbfac821d2c59a61649369e2c20a52aeb59a6620
lightningd(20003): peer 031c527494f295220d26a1814c18e7db3f4f03203ce1a53a0b74b52ed570a3972a: state: ONCHAIND_OUR_UNILATERAL -> FUNDING_SPEND_SEEN
lightning_onchaind(29287): pid 29287, msgfd 24

I don't know if that is the transaction that is trying to post.

I looked for txid 60d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015 in the Bitcoin chain and found nothing.  bitcoin-cli getrawtransaction 60d4f5831b1230a2c5ea0378ce8884232fc6d58cfbc7cedccaab49bf2208c015 showed the transaction, and I see it has 300 confirmations.

Now, I go back to my other peer log and see (older logs first, most recent log after most recent git rebuild and restart):

...
95->96 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log.1:lightningd(16747): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got depth change 96->97 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log.1:lightningd(16747): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got depth change 97->98 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log.1:lightningd(16747): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got depth change 98->99 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log.1:lightningd(16747): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got depth change 99->100 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log:lightningd(14787): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got UTXO spend for 15c00822bf49abcadccec7fb8cd5c62f238488ce7803eac5a230121b83f5d460:0: 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180
log/log:lightning_onchaind(21950): TRACE: Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by THEIR_UNILATERAL (15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180)
log/log:lightning_onchaind(21950): TRACE: Tracking output 0 of 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180: THEIR_UNILATERAL/DELAYED_OUTPUT_TO_THEM
log/log:lightning_onchaind(21950): TRACE: Ignoring output 0 of 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180: THEIR_UNILATERAL/DELAYED_OUTPUT_TO_THEM
log/log:lightningd(14787): peer 02a5487cab84e33cacf424633e7520cfb3fce51aeaab41942ab01e72fb49d166f6: Got depth change 0->298 for 15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180

Well, that looks better!  Let's see:

"lightning-cli listfunds" in my better node shows just the funds I already had loaded in it, whereas "lightning-cli listfunds" in my other node that had nothing before now shows:

ulmo@ubcln:~/log$ lc listfunds
{ "outputs" : 
        [ 
                { "txid" : "9b8430ea0ee02a5a0ff92fdef09e01ecba0fbf5020517d6ea39f0354e9f8eea0", "output" : 0, "value" : 9909585 }, 
                { "txid" : "174d9986793545108d26cf3c06927a2a68c55a9ade42f3c2bcdf471abe93cddd", "output" : 0, "value" : 9909585 } ] }
ulmo@ubcln:~/log$ 

Much better!  Are any of those recent, and where did they come from?  Looks like yes, the second one:

Looking at blockchain.info for both of those, I see that the output of the transaction I mentioned before (15b98c08c0bd5b2a3adf08ffefc9aac7b63bbdd4a89211979da515174fbf5180) is bc1qc9kgncsuud9gvy2z4r5q7gpalj86h7tlksqhx39gud2gvmxax4qsnvufhl, and the transaction above with 2 confirmations txid 174d9986793545108d26cf3c06927a2a68c55a9ade42f3c2bcdf471abe93cddd has input bc1qc9kgncsuud9gvy2z4r5q7gpalj86h7tlksqhx39gud2gvmxax4qsnvufhl, which is the same output as the one I was ballyhooing about above.

As an added bonus, apparently my tracking down what happened to my money arrived at not one but two latent transactions getting posted.  That's great!

Thank you very much.
